tests/components/climate/test_init.py

```diff
@@ -378,6 +378,29 @@ def supported_features(self) -> int:
     assert "is using deprecated supported features values" not in caplog.text
 
 
+def test_deprecated_supported_features_ints_with_on(
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test deprecated supported features ints with on."""
+    int_value = 1 | 128 | 256
```

Comment:
I think we could parametrize this (instead of basically duplicating the above test)?

Reply:
Adjusted and used `.value` now so its a bit more clear what is being tested